### ISSUE TITLE
[Request Feature] reinitialize / reinit mutation

### ISSUE THREAD

**Issue by Jekiwijaya** [marcin-piela/react-fetching-library#47]
After calling mutate, I need to reinit if there's some error. Is it currently possible? I don't find a method that satisfies that condition

**Comment by marcin-piela** [comment]
What do you mean by `reinit`? Could you provide some example?

**Comment by Jekiwijaya** [comment]
something like this: https://noelyoo.github.io/use-async-fn

After i call `mutate` function, and if my API has some error, the variable `error` that given by `useMutation` would be true. I want give user a button to reset the form, but I need to make the variable `error` to be false again, also the `payload`. By I mean reinit is reinitialize the useMutation state to initialState value.

your response is so fast, thank you 🙇

**Comment by marcin-piela** [comment]
Unfortunately there is no such option, we can add it in next release (for all hooks) cause it can be useful in some cases.

By the way we use `final-form` and there is no need to reset `mutation`, cause errors are passed during submit to form and we can reset form.

**Comment by marcin-piela** [comment]
Released in version 1.5.5

Example:
```js
const { reset } = useQuery(action);
const { reset: resetMutation} = useMutation((values) => action(values));

return <>
    <button onClick={reset}>Reset query state</button>
    <button onClick={resetMutation}>Reset mutation state</button>
</>
```
